mycore-restapi/src/main/java/org/mycore/restapi/converter/MCRContentAbstractWriter.java

```diff
@@ -0,0 +1,88 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.restapi.converter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+
+import org.apache.logging.log4j.LogManager;
+import org.mycore.common.content.MCRContent;
+import org.mycore.common.content.transformer.MCRContentTransformer;
+import org.mycore.common.content.transformer.MCRContentTransformerFactory;
+import org.mycore.restapi.annotations.MCRParam;
+import org.mycore.restapi.annotations.MCRParams;
+
+public abstract class MCRContentAbstractWriter implements MessageBodyWriter<MCRContent> {
+    public static final String PARAM_OBJECTTYPE = "objectType";
+
+    private static Optional<String> getTransformerId(Annotation[] annotations, String format, String detail) {
+        return Stream.of(annotations)
+            .filter(a -> MCRParams.class.isAssignableFrom(a.annotationType()))
+            .map(MCRParams.class::cast)
+            .flatMap(p -> Stream.of(p.values()))
+            .filter(p -> p.name().equals(PARAM_OBJECTTYPE))
+            .findAny()
+            .map(MCRParam::value)
+            .map(objectType -> objectType + "-" + format + "-" + detail);
+    }
+
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return getTransfomerFormat().isCompatible(mediaType) && MCRContent.class.isAssignableFrom(type);
+    }
+
+    @Override
+    public void writeTo(MCRContent content, Class<?> type, Type genericType, Annotation[] annotations,
+        MediaType mediaType, MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream)
+        throws IOException, WebApplicationException {
+        String format = getTransfomerFormat().getSubtype();
+        String detail = mediaType.getParameters().getOrDefault(MCRDetailLevel.MEDIA_TYPE_PARAMETER,
+            MCRDetailLevel.normal.toString());
+        LogManager.getLogger().debug("MediaType={}, format={}, detail={}", mediaType, format, detail);
+        Optional<String> transformerId = getTransformerId(annotations, format, detail);
+        LogManager.getLogger().debug("Transformer for {} would be {}.", content.getSystemId(),
+            transformerId.orElse(null));
+        Optional<MCRContentTransformer> transformer = transformerId
+            .map(MCRContentTransformerFactory::getTransformer);
+        if (transformer.isPresent()) {
```

Comment:
I think have to put a note to the documentation that the transformer should not access the database, because most read operations dont have MCRRequireTransaction.

Reply:
I do not know since when, but transaction are only **required** if you write to database and **can be used** if you perform read operations. All of "/v2/*" does act like that.